src/skmatter/clustering/_quick_shift.py

```diff
@@ -108,7 +108,7 @@ def fit(self, X, y=None, samples_weight=None):
 
         if (self.cell is not None) and (X.shape[1] != len(self.cell)):
             raise ValueError("Cell dimension does not match the data dimension.")
-        dist_matrix = DIST_METRICS[self.metric](X, X, squared=True, cell=self.cell)
+        dist_matrix = self.metric(X, X, squared=True, cell=self.cell)
```

Comment:
I think you have to do something like 

```suggestion
        dist_matrix = self.metric(X, X, squared=True, **self.metric_params)
```

and check convert `self.metric_params` into a dict if this is `None`.

When I need a callable and extra params for a class I usually define something like this in the `__init__`

```python
self.metric = lambda X, Y, squared: metric(X, Y,squared,** metric_params)
```

the lambda function already absorbed the extra arguments and later I can call the function with the same syntac regardless what the input callable was:

```python
dist_matrix = self.metric(X, X, squared=True)
```